### PR TITLE
[GEN-515] Live Climb Tracker

### PR DESCRIPTION
**Pull request by abhipi**
- Added button next to Developer Feedback to navigate to the Climb Tracker component (will fix UX in next PR).
- New logic file for the Live activity tracker (checked runtime, accounted for loading wheel). 
- Minimized loading time with new Firebase query in TapsApi (built a firebase index for it too).
- Added the graph, live line, and text based on how busy (quartile based). 
- Images attached below.

image
image

### CODE REVIEW

File: trk/src/Navigation/AppNavigator.js [hunk 1]
```diff
@@ -86,10 +105,15 @@ function HomeStack() {
           title: 'Home',
           headerBackTitleVisible: null,
           headerRight: () => (
+            <View style={{display: 'flex', flexDirection: 'row'}}>
+            <TrackerButton 
+            title="Tracker"
+            navigation={navigation}/>
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:111]: keep

File: trk/src/Navigation/AppNavigator.js [hunk 2]
```diff
@@ -98,6 +122,11 @@ function HomeStack() {
         component={DeveloperFeedbackForm}
         options={{ title: 'Developer Feedback', headerTitleAlign: 'center' }}
       />
+      <Stack.Screen
+        name="Climbs_Tracker"
+        component={LiveClimbTracker}
+        options={{ title: 'Climb Tracker', headerTitleAlign: 'center' }}
+      />
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:129]: keep

File: trk/src/Navigation/AppNavigator.js [hunk 3]
```diff
@@ -128,10 +157,15 @@ function ProfileStack() {
           title: 'Profile',
           headerBackTitleVisible: false,
           headerRight: () => (
+            <View style={{display: 'flex', flexDirection: 'row'}}>
+            <TrackerButton 
+            title="Tracker"
+            navigation={navigation}/>
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:163]: no need

File: trk/src/Navigation/AppNavigator.js [hunk 4]
```diff
@@ -140,6 +174,11 @@ function ProfileStack() {
         component={DeveloperFeedbackForm}
         options={{ title: 'Developer Feedback', headerTitleAlign: 'center' }}
       />
+      <Stack.Screen
+        name="Climbs_Tracker"
+        component={LiveClimbTracker}
+        options={{ title: 'Climb Tracker', headerTitleAlign: 'center' }}
+      />
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:181]: no need

File: trk/src/Navigation/AppNavigator.js [hunk 5]
```diff
@@ -183,10 +222,15 @@ function AnalyticsStack() {
           title: 'My Gym',
           headerBackTitleVisible: false,
           headerRight: () => (
+            <View style={{display: 'flex', flexDirection: 'row'}}>
+            <TrackerButton 
+            title="Tracker"
+            navigation={navigation}/>
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:228]: no need

File: trk/src/Navigation/AppNavigator.js [hunk 6]
```diff
@@ -195,6 +239,11 @@ function AnalyticsStack() {
         component={DeveloperFeedbackForm}
         options={{ title: 'Developer Feedback', headerTitleAlign: 'center' }}
       />
+      <Stack.Screen
+        name="Climbs_Tracker"
+        component={LiveClimbTracker}
+        options={{ title: 'Climb Tracker', headerTitleAlign: 'center' }}
+      />
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:246]: no need

File: trk/src/Navigation/AppNavigator.js [hunk 7]
```diff
@@ -214,10 +263,15 @@ function ClimbInputStackScreen() {
         component={ClimbInputData}
         options={({ navigation }) => ({
           headerRight: () => (
+            <View style={{display: 'flex', flexDirection: 'row'}}>
+            <TrackerButton 
+            title="Tracker"
+            navigation={navigation}/>
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:269]: no need

File: trk/src/Navigation/AppNavigator.js [hunk 8]
```diff
@@ -226,6 +280,11 @@ function ClimbInputStackScreen() {
         component={DeveloperFeedbackForm}
         options={{ title: 'Developer Feedback', headerTitleAlign: 'center' }}
       />
+      <Stack.Screen
+        name="Climbs_Tracker"
+        component={LiveClimbTracker}
+        options={{ title: 'Climb Tracker', headerTitleAlign: 'center' }}
+      />
```
Comment (VGmt) [inline trk/src/Navigation/AppNavigator.js:287]: no need

File: trk/src/Screens/LiveClimbTracker/index.js [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+    const [openClimbPicker, setOpenClimbPicker] = useState(false);
+    const currentDate = moment().tz('America/New_York');
+    const [activeDate, setActiveDate] = useState(currentDate);
+    
+    const [reloadTrigger, setReloadTrigger] = useState(false);
+    const handleReload = () => {
+        setReloadTrigger(prev => !prev); // Toggle the state to trigger a reload
+    };
+
+    const [selectedGymId, setSelectedGymId] = useState(null);
```
Comment (VGmt) [inline trk/src/Screens/LiveClimbTracker/index.js:28]: The only gym we want users to get info on is Movement LIC. So IMO you can put it as default, and only choosable value

Comment (VGmt) [reply trk/src/Screens/LiveClimbTracker/index.js:28]: for now... ;)

In the future, we'll allow users to select more than one, once we have more gyms. Though, we'll want to restrict users' access to only gyms they subscribe to. Not there yet, but soon™

File: trk/src/Screens/LiveClimbTracker/index.js [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+                    setLiveLineText("Very Busy");
+                }
+            }
+    
+            const percentOfDayPassed = (currentTime.diff(chartStartTime) / chartEndTime.diff(chartStartTime));
+            setLiveLinePosition(percentOfDayPassed * chartWidth);
+        } else {
+            setLiveLinePosition(null);
+        }
+    };
```
Comment (VGmt) [inline trk/src/Screens/LiveClimbTracker/index.js:90]: 🚀

**Review by VGmt — CHANGES_REQUESTED** [review]
There's some strong coding skills in there. Keep up the good work 💪.

There's a couple changes needed. We only need the live traffic data on the home screen.  Not on other tabs. I've annotated the bits where it can be removed.

**Review by VGmt — CHANGES_REQUESTED** [review]
There's some strong coding skills in there. Keep up the good work 💪.

There's a couple changes needed. We only need the live traffic data on the home screen.  Not on other tabs. I've annotated the bits where it can be removed.

**Review by redpepper-nag — APPROVED** [review]
This is amazing work Abhishek! No other changes to add aside from Victor's notes

**Pull request merged**